File: cs/Problems/Day9Test.cs
namespace aoc24.Problems;

public class Day9Test
{
    private readonly Day9 sut = new();

    [Theory, InlineData(1928)]
    public void TestSet_ShouldYield_Result(long expected)
    {
        var input = InputReader.ReadProblemInput("day9_1");
        var result = sut.Solve(input);

        Assert.Equal(expected, result);
    }

    [Theory, InlineData(6242766523059)]
    public void FullSet_ShouldYield_Result(long expected)
    {
        var input = InputReader.ReadProblemInput("day9_2");
        var result = sut.Solve(input);

        Assert.Equal(expected, result);
    }
}

File: go/problems/day9_test.go
package problems

import "testing"

func TestDay9_TestSet_YieldsExpectedResult(t *testing.T) {
	TestAocProblem(t, Day9{}, "day9_1", 1928)
}

func TestDay9_FullSet_YieldsExpectedResult(t *testing.T) {
	TestAocProblem(t, Day9{}, "day9_2", 6242766523059)
}
